dev-lang/zig/zig-0.13.0.ebuild

```diff
@@ -0,0 +1,211 @@
+# You can check https://github.com/ziglang/zig-bootstrap in future, to see
+# options that are passed to LLVM CMake building (excluding "static" ofc).
+DEPEND="
+	sys-devel/clang:${LLVM_MAX_SLOT}=
+	sys-devel/lld:${LLVM_MAX_SLOT}=
+	sys-devel/llvm:${LLVM_MAX_SLOT}=[zstd]
+"
+
+RDEPEND="
+	${DEPEND}
+"
+
+IDEPEND="app-eselect/eselect-zig"
+
+# see https://github.com/ziglang/zig/issues/3382
+# For now, Zig Build System doesn't support enviromental CFLAGS/LDFLAGS/etc.
+QA_FLAGS_IGNORED="usr/.*/zig/${PV}/bin/zig"
+
+# Since commit https://github.com/ziglang/zig/commit/e7d28344fa3ee81d6ad7ca5ce1f83d50d8502118
+# Zig uses self-hosted compiler only
+CHECKREQS_MEMORY="4G"
+
+PATCHES=(
+	"${FILESDIR}/${P}-test-fmt-no-doc.patch"
+	"${FILESDIR}/${P}-test-std-kernel-version.patch"
+	"${FILESDIR}/${P}-fix-dir-makepath.patch"
+)
+
+llvm_check_deps() {
+	has_version "sys-devel/clang:${LLVM_SLOT}"
+}
+
+ctarget_to_zigtarget() {
+	# Zig's Target Format: arch-os-abi
+	local CTARGET="${CTARGET:-${CHOST}}"
+
+	local ZIG_ARCH
+	case "${CTARGET%%-*}" in
+		i?86)		ZIG_ARCH=x86;;
+		sparcv9)	ZIG_ARCH=sparc64;;
+		*)		ZIG_ARCH="${CTARGET%%-*}";; # Same as in CHOST
+	esac
+
+	local ZIG_OS
+	case "${CTARGET}" in
+		*linux*)	ZIG_OS=linux;;
+		*apple*)	ZIG_OS=macos;;
+	esac
+
+	local ZIG_ABI
+	case "${CTARGET##*-}" in
+		gnu)		ZIG_ABI=gnu;;
+		solaris*)	ZIG_OS=solaris ZIG_ABI=none;;
+		darwin*)	ZIG_ABI=none;;
+		*)		ZIG_ABI="${CTARGET##*-}";; # Same as in CHOST
+	esac
+
+	echo "${ZIG_ARCH}-${ZIG_OS}-${ZIG_ABI}"
+}
+
+get_zig_mcpu() {
+	local ZIG_DEFAULT_MCPU=native
+	tc-is-cross-compiler && ZIG_DEFAULT_MCPU=baseline
+	echo "${ZIG_MCPU:-${ZIG_DEFAULT_MCPU}}"
+}
+
+get_zig_target() {
+	local ZIG_DEFAULT_TARGET=native
+	tc-is-cross-compiler && ZIG_DEFAULT_TARGET="$(ctarget_to_zigtarget)"
+	echo "${ZIG_TARGET:-${ZIG_DEFAULT_TARGET}}"
+}
+
+pkg_setup() {
+	llvm_pkg_setup
+	check-reqs_pkg_setup
+}
+
+src_configure() {
+	# Useful for debugging and a little bit more deterministic.
+	export ZIG_LOCAL_CACHE_DIR="${T}/zig-local-cache"
+	export ZIG_GLOBAL_CACHE_DIR="${T}/zig-global-cache"
+
+	local mycmakeargs=(
+		-DZIG_SHARED_LLVM=ON
+		-DZIG_TARGET_TRIPLE="$(get_zig_target)"
+		-DZIG_TARGET_MCPU="$(get_zig_mcpu)"
+		-DZIG_USE_LLVM_CONFIG=ON
+		-DCMAKE_PREFIX_PATH="$(get_llvm_prefix ${LLVM_MAX_SLOT})"
+		-DCMAKE_INSTALL_PREFIX="${EPREFIX}/usr/$(get_libdir)/zig/${PV}"
+	)
+
+	cmake_src_configure
+}
+
+src_compile() {
+	cmake_src_compile
+
+	"${BUILD_DIR}/stage3/bin/zig" env || die "Zig compilation failed"
+
+	if use doc; then
+		cd "${BUILD_DIR}" || die
+		edo ./stage3/bin/zig build std-docs --prefix "${S}/docgen/"
+		edo ./stage3/bin/zig build langref --prefix "${S}/docgen/"
+	fi
+}
+
+src_test() {
+	cd "${BUILD_DIR}" || die
+	local ZIG_TEST_ARGS=(
+		-j$(makeopts_jobs)
+		--color on
+		--summary all
+		--verbose
+		-Dstatic-llvm=false
+		-Denable-llvm
+		-Dskip-non-native
+		-Doptimize=Debug
+		-Dtarget="$(get_zig_target)"
+		-Dcpu="$(get_zig_mcpu)"
+	)
+	local ZIG_TEST_STEPS=(
```

Comment:
I'd include a note about dropping some test steps in the commit message, I understand why you dropped test-fmt. But it may not be obvious from the diff that its even missing.

Reply:
I haven't dropped any tests. I simply formatted and sorted the list.

I believe having such formatting is better when we have to review updates, so we will benefit from it in the future.

Sorry, didn't intend to confuse!